godot-kotlin/godot-library/src/nativeCore/kotlin/godot/core/type/Vector2.kt

```diff
@@ -0,0 +1,434 @@
+     */
+    fun sign(): Vector2 {
+        return Vector2(sign(x), sign(y))
+    }
+
+    /**
+     * Returns the result of spherical linear interpolation between this vector and b, by amount t.
+     * t is in the range of 0.0 - 1.0, representing the amount of interpolation.
+     *
+     * Note: Both vectors must be normalized.
+     */
+    fun slerp(b: Vector2, t: RealT): Vector2 {
+        if (!this.isNormalized() || !b.isNormalized()) {
+            Godot.printError("Vectors not normalized", "slerp()", "Vector2.kt", 240)
+        }
+        val theta: RealT = angleTo(b)
+        return rotated((theta * t))
+    }
+
+    /**
+     * Returns the component of the vector along a plane defined by the given normal.
+     */
+    fun slide(vec: Vector2): Vector2 {
+        return vec - this * this.dot(vec)
+    }
+
+    /**
+     * Returns the vector snapped to a grid with the given size.
+     */
+    fun snapped(by: Vector2): Vector2 {
+        val newX: RealT = if (by.x != 0.0) floor(x / by.x + 0.5) else x
+        val newY = if (by.y != 0.0) floor(y / by.y + 0.5) else y
+        return Vector2(newX, newY)
+    }
+
+    /**
+     * Returns a perpendicular vector.
+     */
+    fun tangent(): Vector2 {
+        return Vector2(y, -x)
+    }
+
+
+    //UTILITIES
+    operator fun get(idx: Int): RealT =
+        when (idx) {
+            0 -> x
+            1 -> y
+            else -> throw IndexOutOfBoundsException()
+        }
+
+    operator fun set(n: Int, f: RealT) =
+        when (n) {
+            0 -> x = f
+            1 -> y = f
+            else -> throw IndexOutOfBoundsException()
+        }
+
+    operator fun plus(v: Vector2): Vector2 =
+        Vector2(x + v.x, y + v.y)
+
+    operator fun minus(v: Vector2): Vector2 =
+        Vector2(x - v.x, y - v.y)
+
+    operator fun times(v1: Vector2): Vector2 =
+        Vector2(x * v1.x, y * v1.y)
+
+    operator fun times(rvalue: RealT): Vector2 =
+        Vector2(x * rvalue, y * rvalue)
+
+    operator fun div(v1: Vector2): Vector2 =
+        Vector2(x / v1.x, y / v1.y)
+
+    operator fun div(rvalue: RealT): Vector2 =
+        Vector2(x / rvalue, y / rvalue)
+
+    operator fun unaryMinus(): Vector2 =
+        Vector2(-x, -y)
+
+
+    override fun equals(other: Any?): Boolean =
+        when (other) {
+            is Vector2 -> (x == other.x && y == other.y)
+            else -> false
+        }
+
+    override fun compareTo(other: Vector2): Int =
+        if (x == other.x) {
+            when {
+                y < other.y -> -1
+                y == other.y -> 0
+                else -> 1
+            }
+        } else {
+            when {
+                x < other.x -> -1
+                else -> 1
+            }
+        }
+
+    override fun toString(): String {
+        return "($x, $y)"
+    }
+
+    override fun hashCode(): Int {
+        return this.toString().hashCode()
+    }
+}
+
+operator fun RealT.times(vec: Vector2) = vec * this
```

Comment:
Why do we need this?

Reply:
it's just the counterpart of     
```
operator fun times(rvalue: RealT): Vector2 =
        Vector2(x * rvalue, y * rvalue)
```
That way we can write: 2.0 * vec or vec * 2.0 regardless of the order